packages/math-input/src/components/key-handlers/key-translator.ts

```diff
@@ -43,72 +43,75 @@ function buildGenericCallback(
     };
 }
 
-function buildNormalFunctionCallback(command: string) {
+function buildNormalFunctionCallback(command: string, translated?: string) {
     return function (mathField: MathFieldInterface) {
-        mathField.write(`\\${command}\\left(\\right)`);
+        mathField.write(`\\${translated}\\left(\\right)`);
         mathField.keystroke("Left");
     };
 }
 
-const keyToMathquillMap: Record<Key, MathFieldUpdaterCallback> = {
-    EXP: handleExponent,
-    EXP_2: handleExponent,
-    EXP_3: handleExponent,
-
-    JUMP_OUT_PARENTHESES: handleJumpOut,
-    JUMP_OUT_EXPONENT: handleJumpOut,
-    JUMP_OUT_BASE: handleJumpOut,
-    JUMP_INTO_NUMERATOR: handleJumpOut,
-    JUMP_OUT_NUMERATOR: handleJumpOut,
-    JUMP_OUT_DENOMINATOR: handleJumpOut,
-
-    LEFT: handleArrow,
-    RIGHT: handleArrow,
-
-    LOG: buildNormalFunctionCallback("log"),
-    LN: buildNormalFunctionCallback("ln"),
-    SIN: buildNormalFunctionCallback("sin"),
-    COS: buildNormalFunctionCallback("cos"),
-    TAN: buildNormalFunctionCallback("tan"),
-
-    CDOT: buildGenericCallback("\\cdot"),
-    DECIMAL: buildGenericCallback(decimalSymbol),
-    DIVIDE: buildGenericCallback("\\div"),
-    EQUAL: buildGenericCallback("="),
-    GEQ: buildGenericCallback("\\geq"),
-    GT: buildGenericCallback(">"),
-    LEQ: buildGenericCallback("\\leq"),
-    LT: buildGenericCallback("<"),
-    MINUS: buildGenericCallback("-"),
-    NEGATIVE: buildGenericCallback("-"),
-    NEQ: buildGenericCallback("\\neq"),
-    PERCENT: buildGenericCallback("%"),
-    PERIOD: buildGenericCallback("."),
-    PLUS: buildGenericCallback("+"),
-    TIMES: buildGenericCallback("\\times"),
+const keyToMathquillMap: Record<Key, (translated: string) => MathFieldUpdaterCallback> = {
+    EXP: () => handleExponent,
+    EXP_2: () => handleExponent,
+    EXP_3: () => handleExponent,
+
+    JUMP_OUT_PARENTHESES: () => handleJumpOut,
+    JUMP_OUT_EXPONENT: () => handleJumpOut,
+    JUMP_OUT_BASE: () => handleJumpOut,
+    JUMP_INTO_NUMERATOR: () => handleJumpOut,
+    JUMP_OUT_NUMERATOR: () => handleJumpOut,
+    JUMP_OUT_DENOMINATOR: () => handleJumpOut,
+
+    LEFT: () => handleArrow,
+    RIGHT: () => handleArrow,
+
+    LOG: () => buildNormalFunctionCallback("log"),
+    LN: () => buildNormalFunctionCallback("ln"),
+    SIN: (translated?: string) => {
+        console.log(`\\${translated}\\left(\\right)`)
+        return buildNormalFunctionCallback("sin", translated)
```

Comment:
this is where we mix in the translated value and map a "sen" keypad press to a "sen" in the math input keypad.